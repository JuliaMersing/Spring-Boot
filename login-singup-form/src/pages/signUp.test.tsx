import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SignUp from './signUp';

const mockUseDispatch = jest.fn();
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockUseDispatch,
}));

describe('signUp component', () => {
  const setUp = () => {
    const utils = render(<SignUp />);

    const signUpButton = screen.getByRole('button');
    const rememberCheck = screen.getByRole('checkbox');
    return {
      ...utils,
      signUpButton,
      rememberCheck,
    };
  };
  test('should render component and match snapshot', () => {
    const { container } = render(<SignUp />);
    expect(container)
      .toMatchSnapshot();
  });
  test(
    'should call onSubmit when button is clicked',
    async () => {
      const {
        signUpButton,
      } = setUp();

      await userEvent.click(signUpButton);

      expect(mockUseDispatch)
        .toHaveBeenCalledTimes(1);
    },
  );
});
