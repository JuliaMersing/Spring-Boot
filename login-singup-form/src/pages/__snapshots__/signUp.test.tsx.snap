// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp component should render component and match snapshot 1`] = `
<div>
  <div
    class="min-h-full h-screen flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8"
  >
    <div
      class="max-w-md w-full space-y-8"
    >
      <div
        class="mb-10"
      >
        <div
          class="flex justify-center"
        >
          <i
            class="fa-solid fa-bell-concierge text-[56px] text-purple-600 justify-center"
          />
        </div>
        <h2
          class="mt-6 text-center text-3xl font-extrabold text-gray-900"
        >
          Sign Up to create an account
        </h2>
        <p
          class="mt-2 text-center text-sm text-gray-600 mt-5"
        >
          Already have an account?
          <a
            class="font-medium text-purple-600 hover:text-purple-500 ml-1"
            href="/"
          >
            Login
          </a>
        </p>
      </div>
      <form
        data-testid="login-form"
      >
        <div
          class="mb-6"
        >
          <input
            class="input"
            data-testid="email-input"
            id="email"
            placeholder="Email address"
            type="email"
            value=""
          />
        </div>
        <div
          class="mb-6"
        >
          <input
            class="input"
            data-testid="password-input"
            id="password"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div
          class="mb-6"
        >
          <input
            class="input"
            data-testid="password-input"
            id="password"
            placeholder="Confirm password"
            type="password"
            value=""
          />
        </div>
        <div
          class="flex-space-between"
        >
          <div
            class="flex-center "
          >
            <input
              checked=""
              class="checkbox"
              type="checkbox"
            />
            <p
              class="text-gray-800 ml-2"
            >
              Remember me 
            </p>
          </div>
        </div>
        <div
          class="text-center-left"
        >
          <button
            class="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-purple-600 hover:bg-purple-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500 mt-10"
            type="submit"
          >
             Sign Up 
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
